File: dart/scrabble-score/lib/scrabble_score.dart
// Put your code here
Map<String, int> scores = {
  'AEIOULNRST': 1,
  'DG': 2,
  'BCMP': 3,
  'FHVWY': 4,
  'K': 5,
  'JX': 8,
  'QZ': 10
};

int score(String text) {
  int sum = 0;

  text.toUpperCase().split("").forEach((element) {
    sum += getLetterScore(element);
  });

  return sum;
}

int getLetterScore(String letter) {
  String key = scores.keys.firstWhere((element) => element.contains(letter));
  return scores[key];
}

File: typescript/scrabble-score/scrabble-score.ts
const SCRABLE_SCORES = [
  {
    letters: 'AEIOULNRST',
    score: 1,
  },
  {
    letters: 'DG',
    score: 2,
  },
  {
    letters: 'BCMP',
    score: 3,
  },
  {
    letters: 'FHVWY',
    score: 4,
  },
  {
    letters: 'K',
    score: 5,
  },
  {
    letters: 'JX',
    score: 8,
  },
  {
    letters: 'QZ',
    score: 10,
  },
];

export default function score(text: string | undefined) {
  if (!text) {
    return 0;
  }

  return [...text.toUpperCase()].reduce(
    (total, el) => total + letterScore(el),
    0
  );
}

function letterScore(letter: string): number {
  return SCRABLE_SCORES.find((x) => x.letters.includes(letter))?.score ?? 0;
}
